core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -847,56 +846,64 @@ func loadEthTxes(q pg.Queryer, attempts []EthTxAttempt) error {
 // FindEthTxsRequiringRebroadcast returns attempts that hit insufficient eth,
 // and attempts that need bumping, in nonce ASC order
 func FindEthTxsRequiringRebroadcast(ctx context.Context, q pg.Q, lggr logger.Logger, address gethCommon.Address, blockNum, gasBumpThreshold, bumpDepth int64, maxInFlightTransactions uint32, chainID big.Int) (etxs []*EthTx, err error) {
-	// NOTE: These two queries could be combined into one using union but it
-	// becomes harder to read and difficult to test in isolation. KISS principle
-	etxInsufficientEths, err := FindEthTxsRequiringResubmissionDueToInsufficientEth(ctx, q, lggr, address, chainID)
-	if err != nil {
-		return nil, err
-	}
+	stmt := `
+		SELECT (eth_txes).*, ARRAY_AGG(attempt ORDER BY (attempt).created_at) AS eth_tx_attempts
+		FROM (
+			SELECT
+				ROW(t.*)::eth_txes AS eth_txes,
+				ROW(a.*)::eth_tx_attempt AS attempt,
+				ROW_NUMBER() OVER(ORDER BY nonce) AS depth
+			FROM eth_txes AS t LEFT JOIN eth_tx_attempts AS a ON
+				t.id = a.eth_tx_id AND (a.state != 'broadcast' OR (broadcast_before_block_num IS NOT NULL AND $4 < broadcast_before_block_num))
+			WHERE t.state = 'unconfirmed' AND t.from_address = $1 AND t.evm_chain_id = $2
+		) x
+		WHERE (attempt).state = 'insufficient_eth'::eth_tx_attempts_state OR $3 = 0 OR depth <= $3 GROUP BY eth_txes ORDER BY nonce`
```

Comment:
I'm nervous at the additional complexity introduced by this query. Per-table selects do generate more queries but they are also straightforward, easy to understand and perform well.

Do we have any flamegraph or profiling analysis that indicates these queries were really a bottleneck?